src/industry_cmd.cpp

```diff
@@ -1544,18 +1548,17 @@ static bool CheckCanTerraformSurroundingTiles(TileIndex tile, uint height, int i
  * This function tries to flatten out the land below an industry, without
  *  damaging the surroundings too much.
  */
-static bool CheckIfCanLevelIndustryPlatform(TileIndex tile, DoCommandFlag flags, const IndustryTileTable *it, int type)
+static bool CheckIfCanLevelIndustryPlatform(TileIndex tile, DoCommandFlag flags, const IndustryTileLayout &layout, int type)
 {
-	const int MKEND = -0x80;   // used for last element in an IndustryTileTable (see build_industry.h)
 	int max_x = 0;
 	int max_y = 0;
 
 	/* Finds dimensions of largest variant of this industry */
-	do {
-		if (it->gfx == 0xFF) continue;  //  FF been a marquer for a check on clear water, skip it
+	for (IndustryTileLayout::const_iterator it = layout.begin(); it != layout.end(); ++it) {
+		if (it->gfx == GFX_WATERTILE_SPECIALCHECK) continue;  // marker for a check for clear water, skip it
```

Comment:
previously mentioned about not using range for to avoid changing the loop body, but since you are here anyway... ;)

also, comment is now useless - remove